### PR TITLE
docs(core): improvements to branching wizard focus/selection logic

### LINKED ISSUE #5808: Bug: (core) Wizard - When you try to select another radio button and click 'Cancel' in confirmation pop-up, focus disappears
#### Is this a bug, enhancement, or feature request?
Bug
#### Briefly describe your proposal.
Wizard component, branching example:
When you selected one payment type and would like change it to another - you will get confirmation pop-up. First time if you click 'Cancel' - focus from radio button disappears(no selected radio-buttons), but it should not. Focus should return to previous selected button. Next time (second, third etc.) it will work as expected, focus will be returned to the previous button
#### If this is a bug, please provide steps for reproducing it.
1. Go to https://fundamental-ngx.netlify.app/#/core/wizard#Wizard-Branching
2. Open Wizard
3.  Click 'Go to the payment section page'
4. Select first radio-button
5. Select second radio-button
6. Click 'Cancel' in pop-up
Expected result: focus returned to the first radio-button
Actual result: focus disappeared, no chosen button and when we want to select radio-button we will get pop-up again

Запись_экрана_2021-06-25_в_11 19 04

### PR DESCRIPTION
**Pull request by mikerodonnell89**
#### Please provide a link to the associated issue.
Closes #5808 

#### Please provide a brief summary of this pull request.

Various improvements to the documentation for better handling of selection/deselection/focusing of the payment option radio buttons.

### CODE REVIEW

**Review by Betrozov — APPROVED** [review]

File: apps/docs/src/app/core/component-docs/wizard/examples/wizard-branching-example.component.html [hunk 1]
```diff
@@ -112,7 +112,7 @@
 </div>
 
 <ng-template let-dialog let-dialogConfig="dialogConfig" #confirmationDialog>
-    <fd-dialog [dialogConfig]="dialogConfig" [dialogRef]="dialog">
+    <fd-dialog [dialogConfig]="dialogConfig" [dialogRef]="dialog" style="z-index: 1002;">
```
Comment (InnaAtanasova) [inline apps/docs/src/app/core/component-docs/wizard/examples/wizard-branching-example.component.html:115]: Why did you add this? Will the user be required to add z-index in their project? If yes, we need to move this in fund-styles

**Review by N1XUS — APPROVED** [review]

File: apps/docs/src/app/core/component-docs/wizard/examples/wizard-branching-example.component.html (hunk 1 above)
Comment (mikerodonnell89) [reply apps/docs/src/app/core/component-docs/wizard/examples/wizard-branching-example.component.html:115]: @InnaAtanasova Good question, I do not believe this will be necessary for the user to add.  It is because the z-index of the "wizard overlay" class is 1000, and the z-index of the fd-bar in the previous example is 1001 - so this is to ensure the dialog is over top of both the wizard overlay and the buttons from the fd-bar in the "mobile wizard" examples

**Review by InnaAtanasova — APPROVED** [review]
